disco_aws_automation/disco_deploy.py

```diff
@@ -245,48 +235,46 @@ def handle_nodeploy_ami(self, old_hostclass_dict, ami, desired_size, dry_run):
+
+        if not pipeline_dict:
+            pipeline_dict = {"hostclass": hostclass}
 
         logging.info("Smoke testing non-deploy Hostclass %s AMI %s with %s deployment strategy",
-                     old_hostclass_dict["hostclass"], ami.id, DEPLOYMENT_STRATEGY_CLASSIC)
+                     hostclass, ami.id, DEPLOYMENT_STRATEGY_CLASSIC)
 
         if dry_run:
             return
 
-        new_size = desired_size * 2 if desired_size else 1
-        new_hostclass_dict = copy.deepcopy(old_hostclass_dict)
-        new_hostclass_dict["sequence"] = 1
-        new_hostclass_dict["max_size"] = new_size
-        new_hostclass_dict["min_size"] = new_size / 2
-        new_hostclass_dict["desired_size"] = new_size
-        new_hostclass_dict["smoke_test"] = "no"
-        new_hostclass_dict["ami"] = ami.id
-        rollback_hostclass_dict = copy.deepcopy(old_hostclass_dict)
+        new_hostclass_dict = self._generate_deploy_pipeline(
+            deployment_strategy=DEPLOYMENT_STRATEGY_CLASSIC,
+            pipeline_dict=pipeline_dict,
+            old_group=old_group,
+            ami=ami
+        )
+
+        rollback_hostclass_dict = self._generate_final_pipeline(
+            pipeline_dict=pipeline_dict,
+            old_group=old_group,
+            ami=ami
+        )
+        rollback_hostclass_dict["smoke_test"] = "no"
 
         self._disco_aws.spinup([new_hostclass_dict], testing=True)
 
-        rollback_hostclass_dict["sequence"] = 1
-        rollback_hostclass_dict["smoke_test"] = "no"
-        rollback_hostclass_dict["max_size"] = old_hostclass_dict.get("max_size", desired_size)
-        rollback_hostclass_dict["min_size"] = old_hostclass_dict.get("min_size", desired_size)
-        rollback_hostclass_dict["desired_size"] = old_hostclass_dict.get("desired_size", desired_size)
-        rollback_hostclass_dict["desired_size"] = snap_to_range(
-            rollback_hostclass_dict["desired_size"],
-            rollback_hostclass_dict["min_size"], rollback_hostclass_dict["max_size"])
-
-        if self.wait_for_smoketests(ami.id, desired_size or 1):
+        if self.wait_for_smoketests(ami.id, new_hostclass_dict["desired_size"]):
             self._promote_ami(ami, "tested")
-            rollback_hostclass_dict["ami"] = ami.id
         else:
             self._promote_ami(ami, "failed")
             rollback_hostclass_dict.pop("ami", None)
 
-        if rollback_hostclass_dict["desired_size"]:
+        if old_group:
             self._disco_aws.terminate(self._get_new_instances(ami.id), use_autoscaling=True)
             self._disco_aws.spinup([rollback_hostclass_dict])
+            # Create scheduled actions on the ASG.
+            self._create_scaling_schedule(pipeline_dict, hostclass=hostclass)
         else:
-            self._disco_autoscale.delete_groups(hostclass=rollback_hostclass_dict["hostclass"], force=True)
+            self._disco_autoscale.delete_groups(hostclass=hostclass, force=True)
```

Comment:
if an asg doesn't already exist, shouldn't you use the new asg definition and create one?